proposals/0010-validating-resource-container-elements.md

```diff
@@ -0,0 +1,142 @@
+* Proposal: [0010](0010-validating-resource-container-elements.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: TBD
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+*During the review process, add the following fields as needed:*
+
+* PRs: [#NNNN](https://github.com/microsoft/DirectXShaderCompiler/pull/NNNN)
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the RET is `float`, and the resource type is `RWBuffer`. The
+resource type is not a `RawBuffer` variant, and so there is a distinct set
+of rules that define valid RETs for this resource type.

```

Comment:
>  The resource type is not a `RawBuffer` variant

I'm a bit lost here.  I think that there's some context about RawBuffer's that I need to have in order to understand this...but...actually, I can't find any documentation online for `RawBuffer`.  Am I missing something?

Maybe this should be `ByteAddressBuffer`?

And possibly the context here is that ByteAddressBuffers can only read/write `uint`s, whereas `RWBuffer` needs to have a specified element type?

Reply:
I'll add some context in the spec, but rawbuffers are essentially a set of buffers that defer to the shader in interpreting / converting elements in the buffer to be intelligible. Thus, there are fewer constraints on their element types, they don't need to fit in 4 32-bit groups for example. `StructuredBuffer` is an example of a rawbuffer.

Reply:
The thing that's really confusing me here is seeing `RawBuffer` formatted like that - it really suggests that this is the name of something that appears in HLSL code - like `RWBuffer` is.